src/views/Pools/components/PoolCard/Modals/StakeModal.tsx

```diff
@@ -0,0 +1,167 @@
+          TranslateString(999, 'Canceled'),
+          TranslateString(999, 'Please try again and confirm the transaction.'),
+        )
+        setPendingTx(false)
+      }
+    } else {
+      try {
+        // staking
+        await onStake(stakeAmount, stakingToken.decimals)
+        toastSuccess(
+          `${TranslateString(1074, 'Staked')}!`,
+          TranslateString(999, `Your ${stakingToken.symbol} funds have been staked in the pool!`),
+        )
+        setPendingTx(false)
+        onDismiss()
+      } catch (e) {
+        toastError(
+          TranslateString(999, 'Canceled'),
+          TranslateString(999, 'Please try again and confirm the transaction.'),
+        )
+        setPendingTx(false)
+      }
+    }
+  }
+
+  return (
+    <Modal
+      title={isRemovingStake ? TranslateString(588, 'Unstake') : TranslateString(999, 'Stake in Pool')}
+      onDismiss={onDismiss}
+      headerBackground={theme.colors.gradients.cardHeader}
+    >
+      <Flex alignItems="center" justifyContent="space-between" mb="8px">
+        <Text bold>{isRemovingStake ? TranslateString(588, 'Unstake') : TranslateString(316, 'Stake')}:</Text>
+        <Flex alignItems="center" minWidth="70px">
+          <Image src={`/images/tokens/${stakingToken.symbol}.png`} width={24} height={24} alt={stakingToken.symbol} />
+          <Text ml="4px" bold>
+            {stakingToken.symbol}
+          </Text>
+        </Flex>
+      </Flex>
+      <BalanceInput
+        value={stakeAmount}
+        onChange={handleStakeInputChange}
+        currencyValue={`~${usdValueStaked || 0} USD`}
+      />
+      <Text mt="8px" ml="auto" color="textSubtle" fontSize="12px" mb="8px">
+        Balance: {getFullDisplayBalance(stakingMax, stakingToken.decimals)}
+      </Text>
+      <Slider
+        min={0}
+        max={100}
+        value={percent}
+        onValueChanged={handleChangePercent}
+        name="stake"
+        valueLabel={`${percent}%`}
+        step={1}
+      />
+      <Flex alignItems="center" justifyContent="space-between" mt="8px">
+        <StyledButton scale="xs" mx="2px" p="4px 16px" variant="tertiary" onClick={() => handleChangePercent(25)}>
+          25%
+        </StyledButton>
+        <StyledButton scale="xs" mx="2px" p="4px 16px" variant="tertiary" onClick={() => handleChangePercent(50)}>
+          50%
+        </StyledButton>
+        <StyledButton scale="xs" mx="2px" p="4px 16px" variant="tertiary" onClick={() => handleChangePercent(75)}>
+          75%
+        </StyledButton>
+        <StyledButton scale="xs" mx="2px" p="4px 16px" variant="tertiary" onClick={() => handleChangePercent(100)}>
+          MAX
+        </StyledButton>
+      </Flex>
+      <Button
+        isLoading={pendingTx}
+        endIcon={pendingTx ? <AutoRenewIcon spin color="currentColor" /> : null}
+        onClick={handleConfirmClick}
+        mt="24px"
+      >
+        {pendingTx ? TranslateString(802, 'Confirming') : TranslateString(464, 'Confirm')}
```

Comment:
I can confirm staking 0 CAKE. Transaction went fine too, lol. Probably should disable button if 0?

Reply:
Ahaha, yes good point

Reply:
Fixed 👌